subprojects/core/src/main/java/org/gradle/plugin/use/internal/InternalPluginRequest.java

```diff
@@ -17,15 +17,21 @@
 package org.gradle.plugin.use.internal;
 
 import org.gradle.api.Nullable;
-import org.gradle.plugin.internal.PluginId;
+import org.gradle.plugin.management.PluginRequest;
+import org.gradle.plugin.use.PluginId;
 
-public interface PluginRequest {
+public interface InternalPluginRequest extends PluginRequest {
```

Comment:
I think it's necessary unless we want to expose a class to the consumers. If that's what you want then I can do it.